compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1307,6 +1307,26 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         let mut generator = None;
         let mut outer_generator = None;
         let mut next_code = Some(&obligation.cause.code);
+        // By introducing a tuple into the upvar types, the first item in the obligation chain
+        // can be of Tuple type instead of the generator that captured the type. We want to catch for
+        // this case and skip it.
```

Comment:
This does the trick for now, but it's probably not what we want to move forward with. The tuple of upvar type should not be the first item in the obligation chain, probably more in the middle. We probably want to change how the obligation chain is created. I haven't figured out where in the code that is done yet though